### PR TITLE
Possible fix: need for changing patch number for azure

### PR DESCRIPTION
**Pull request by dboun**
- ⚠️ This is untested and I'm not exactly sure how to test it!! (paging @PhucNgo1711 @sarthakpati )
- Only on azure, append the commit hash to the project version. That way it's always unique without the need for developers to manually change it. 
- The local builds outside of azure remain unaffected from this

Potential pitfalls:
- Ugly package release names on azure?
- [CBICA/dcmqi](https://github.com/CBICA/dcmqi/blob/master/azure-pipelines.yml) has a similar azure-pipelines.yml that I'm not sure if it needs to change too
- There is a "change" to ```src/applications/Utilities/dcmqi``` from my PR that was not triggered by me intentionally. Maybe that's how submodules work, but I've never noticed it before.

### CODE REVIEW

**Comment by sarthakpati** [comment]
We should probably discuss this in person before merging.

**Comment by sarthakpati** [comment]
Oh, and maybe add **nonRelease** to the version.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -609,9 +609,18 @@ jobs:
...
+        date=$(date '+%Y.%m.%d')                # date in "YYYY.mm.dd" form
 
+        cd bin
         version=$(grep -i -e "project_version:*" CMakeCache.txt | cut -c24-)
+        ### In nonRelease builds
+        ### append date and git commit hash to the version. 
+        ### This way devs won't have to change the patch number on PRs
+        if [[ $version == *"nonRelease"* ]]; then
+            version=${version}_${date}_${git_hash}
+        fi
```
Comment (dboun) [inline azure-pipelines.yml]: What about this? @sarthakpati @ashishsingh18

Comment (dboun) [reply azure-pipelines.yml]: This checks if nonRelease is in the version and then does the date & hash things.

So basically the workflow should be that nonRelease is always in the cmake version, but when you do a PR for a stable release you remove the nonRelease part and azure builds and releases it normally

**Comment by sarthakpati** [comment]
@PhucNgo1711: would it be better to simply let the installers get created as they are right now and **THEN** rename them using the fix that @dboun is proposing?

**Comment by sarthakpati** [comment]
> @PhucNgo1711: would it be better to simply let the installers get created as they are right now and **THEN** rename them using the fix that @dboun is proposing?
> 
> Basically, add the rename before https://github.com/CBICA/CaPTk/blob/master/azure-pipelines.yml#L174, https://github.com/CBICA/CaPTk/blob/master/azure-pipelines.yml#L412 and https://github.com/CBICA/CaPTk/blob/master/azure-pipelines.yml#L674.

Building on Azure: https://dev.azure.com/sarthakpati/CaPTk/_build/results?buildId=1256&view=results

**Comment by dboun** [comment]
Closed because it's probably not needed

**Comment by sarthakpati** [comment]
Implemented a better solution that is integrated with CMake, thereby ensuring a centralized solution. The branch in question is: https://github.com/sarthakpati/CaPTk/tree/dimitrisAzureCheck

CMake will now display the version of CaPTk getting built right off the bat [[ref](https://github.com/sarthakpati/CaPTk/commit/582b46ddf4d0e47f05c0af4fb8e61f05768693a8)].

**Pull request closed without merging**